codegen/type_converter.go

```diff
@@ -346,16 +346,34 @@ func (c *TypeConverter) genConverterForMap(
 		return err
 	}
 
+	valueStruct, isStruct := toFieldType.ValueSpec.(*compile.StructSpec)
+	sourceIdentifier := fromIdentifier
 	_, isStringKey := toFieldType.KeySpec.(*compile.StringSpec)
+
 	if !isStringKey {
-		return errors.Errorf(
-			"could not convert key (%s), map is not string-keyed.",
-			toField.Name,
-		)
+		realType := compile.RootTypeSpec(toFieldType.KeySpec)
+		switch realType.(type) {
+		case
+			*compile.MapSpec,
+			*compile.SetSpec,
+			*compile.ListSpec:
```

Comment:
We cannot support `*compile.StructSpec` either. 

I'm pretty sure `*compile.Int8Spec` , `*compile.DoubleSpec` and `*compile.BoolSpec` will also fail.